### PR TITLE
feat: configurable oidc userid attribute

### LINKED ISSUE #2281: Make OIDC userid attribute configurable
Currently it defaults to `:sub`.

Probably best to add a separate configuration key for it, `:oidc-userid-attribute` instead of using `:oidc-extra-attributes`.

### PR DESCRIPTION
**Pull request by opqdonut**
fixes #2281

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue
- [x] note if PR is on top of other PR

## Documentation
- [x] update changelog if necessary
- [x] documentation _at least_ for config options (i.e. docs folder)

## Testing
- [ ] complex logic is unit tested
- [ ] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [x] no critical TODOs left to implement

### CODE REVIEW

**Comment by opqdonut** [comment]
marking as WIP, it seems we need more features

**Comment by opqdonut** [comment]
rebased now that #2277 is in

File: src/clj/rems/auth/oidc.clj [hunk 1]
```diff
@@ -45,7 +45,7 @@
         ;; locale – could be used to set preferred lang on first login
         ;; email – non-unique (!) email
         id-data (jwt/validate id-token issuer audience now)
-        identity-base {:eppn (:sub id-data)
+        identity-base {:eppn (get id-data (keyword (getx env :oidc-userid-attribute)))
```
Comment (Macroz) [inline src/clj/rems/auth/oidc.clj:48]: Couldn't the attribute be a keyword instead of a string in the config?

**Review by Macroz — APPROVED** [review]

File: src/clj/rems/auth/oidc.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/auth/oidc.clj:48]: Yeah, it could, but I picked a string so that it would map more directly to the JSON oidc sends us

Comment (opqdonut) [reply src/clj/rems/auth/oidc.clj:48]: One further advantage is that a string works for a key like `":boing foo"`  which can be impossible to express as a keyword literal

**Pull request merged**